packages/vats/src/provisionPool.js

```diff
@@ -0,0 +1,159 @@
+// @jessie-check
+// @ts-check
+
+import { fit, M, makeScalarMapStore } from '@agoric/store';
+import { E, Far } from '@endo/far';
+// TODO: move to narrower package?
+import { observeIteration, observeNotifier } from '@agoric/notifier';
+import { AmountMath, AmountShape } from '@agoric/ertp';
+import { WalletName } from '@agoric/internal';
+import { makeMyAddressNameAdmin, PowerFlags } from './core/basic-behaviors.js';
+
+const { details: X, quote: q } = assert;
+
+const privateShape = harden({
+  poolBank: M.eref(M.remotable('bank')),
+});
+
+/**
+ * Given attenuated access to the funding purse,
+ * handle requests to provision smart wallets.
+ *
+ * @param {(address: string, depositFacet: ERef<DepositFacet>) => Promise<void>} sendInitialPayment
+ */
+const makeBridgeProvisionTool = sendInitialPayment => {
+  /**
+   * @param {{
+   *   bankManager: ERef<BankManager>,
+   *   namesByAddressAdmin: ERef<NameAdmin>,
+   *   walletFactory: ERef<import('@agoric/vats/src/core/startWalletFactory').WalletFactoryStartResult['creatorFacet']>
+   * }} io
+   */
+  const makeHandler = ({ bankManager, namesByAddressAdmin, walletFactory }) =>
+    Far('provisioningHandler', {
+      fromBridge: async (_srcID, obj) => {
+        assert(namesByAddressAdmin, 'no namesByAddressAdmin');
+        assert.equal(
+          obj.type,
+          'PLEASE_PROVISION',
+          X`Unrecognized request ${obj.type}`,
+        );
+        const { address, powerFlags } = obj;
+        console.info('PLEASE_PROVISION', address, powerFlags);
+
+        if (!powerFlags.includes(PowerFlags.SMART_WALLET)) {
+          return;
```

Comment:
Why not error? I haven't looked closely to answer for myself, but would be good yo have an explanation comment here regardless.

Reply:
Not sure. I have a vague memory that I tried that and something bad happened. But I should look more.

Reply:
Now I remember: I was thinking we could register 2 handlers for `BRIDGE_ID.WALLET`, on for `SMART_WALLET` and one for `REMOTE_WALLET`. But now that I think about it, we can only register one handler.